Sync available Jetpack Gutenberg blocks

gititon opened the pull request:
This PR syncs available Jetpack Gutenberg blocks via the Callables sync module/mechanism

#### Testing instructions:

With a working Gutenpack environment, disable the Markdown Jetpack module. You should observe a callable sync (PCYsg-hnH-p2) indicating that the Markdown block is unavailable.

#### Proposed changelog entry for your changes:

No changelog entry needed

gititon committed:
Sync available Gutenberg blocks

gititon committed:
Added test

roccotripaldi reviewed (dismissed):
LGTM! Let's get some upstream validation.

enejb commented:
Nicely done @gititon!

gititon commented:
@Automattic/jetpack-crew can I please get a review when somebody has a chance so I can get this merged? Thank you! 😄

jeherve commented:
@gititon I added it to the 6.8 milestone so we'll get to it before the release.

gititon commented:
Thanks @jeherve!

roccotripaldi commented on sync/class.jetpack-sync-defaults.php:
```diff
@@ -240,6 +240,7 @@ public static function get_constants_whitelist() {
 		'site_icon_url'                    => array( 'Jetpack_Sync_Functions', 'site_icon_url' ),
 		'roles'                            => array( 'Jetpack_Sync_Functions', 'roles' ),
 		'timezone'                         => array( 'Jetpack_Sync_Functions', 'get_timezone' ),
+		'available_jetpack_blocks'         => array( 'Jetpack_Gutenberg', 'get_block_availability' ),
```
I propose we rename this to `available_jetpack_editor_extenstions` or `available_jetpack_editor_assets`
cc: @enejb @lezama 

It's not a blocker per-se, but the callable is actually reporting on the availability or blocks AND plugins. If we are going to re-name for accuracy, we should do so before the release.

brbrr reviewed (dismissed):
Seems to work fine from `jetpack_modules` page. I was able to see `available_jetpack_blocks` sync event in my logs mentioning `[available] => 1` for markdown block. 

Although, deactivating `Markdown` module via Jetpack Settings does not trigger any sync events except `jetpack_deactivate_module`

Also, I noticed that `jetpack_activate_module` is sent twice for single action

roccotripaldi commented:
> Although, deactivating Markdown module via Jetpack Settings does not trigger any sync events except jetpack_deactivate_module

@brbrr - thanks for catching that! we tried to resolve that today, but it turns out it is a lot trickier than we thought. 

We are ok with the functionality as it is. `jetpack_available_blocks` doesn't need to be updated in real time for now - we will try to improve in the next release.

roccotripaldi committed:
no need to register the publicize gutenberg extensions as a block
( it's not a block, it's a plugin. we'll handle it's registration on the client side )

jeherve reviewed (approved):
Merging this now.

jeherve commented:
Cherry-picked to `branch-6.8` in 9e85c912302cc5d1113c036ca86193ab2efe30c3
